controllers/storagecluster/external_resources.go

```diff
@@ -291,6 +303,33 @@ func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) checkSCsAndCallCreateDefaultClaim(instance *ocsv1.StorageCluster) error {
+	cephFsStorageClassName := generateNameForCephFilesystemSC(instance)
+	cephRbdStorageClassName := generateNameForCephBlockPoolSC(instance)
+
+	cephFsStorageClass := &storagev1.StorageClass{}
+	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: cephFsStorageClassName, Namespace: instance.Namespace}, cephFsStorageClass); err != nil {
+		return fmt.Errorf("no storageClass found with name %s. %v", cephFsStorageClassName, err)
+	}
+
+	cephRbdStorageClass := &storagev1.StorageClass{}
+	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: cephRbdStorageClassName, Namespace: instance.Namespace}, cephRbdStorageClass); err != nil {
+		return fmt.Errorf("no storageClass found with name %s. %v", cephRbdStorageClassName, err)
+	}
+
+	cephFsScAnnotations := cephFsStorageClass.GetAnnotations()
+	cephRbdScAnnotations := cephRbdStorageClass.GetAnnotations()
+	if (cephFsScAnnotations == nil || cephFsScAnnotations[StorageClassClaimAnnotation] == "") &&
+		(cephRbdScAnnotations == nil || cephRbdScAnnotations[StorageClassClaimAnnotation] == "") {
+		err := r.createDefaultStorageClassClaims(instance)
+		if err != nil {
+			return fmt.Errorf("failed to created storageClassClaim. %v", err)
+		}
+	}
```

Comment:
Why do we need the checks? why can't we just call `r.createDefaultStorageClassClaims` if these exists?

Reply:
right, the creation handle already exists. changed